src/algebra/algebra/spectrum.lean

```diff
@@ -167,4 +170,69 @@ theorem preimage_units_mul_eq_swap_mul {a b : A} :
   (coe : units R → R) ⁻¹' σ (a * b) = coe ⁻¹'  σ (b * a) :=
 by { ext, exact unit_mem_mul_iff_mem_swap_mul, }
 
+end scalar_ring
+
+section scalar_field
+
+variables {𝕜 : Type u} {A : Type v}
+variables [field 𝕜] [ring A] [algebra 𝕜 A]
+
+local notation `σ` := spectrum 𝕜
+local notation `↑ₐ` := algebra_map 𝕜 A
+
+/-- Without the assumption `nontrivial A`, then `0 : A` would be invertible. -/
+@[simp] lemma zero_eq [nontrivial A] : σ (0 : A) = {0} :=
+begin
+  apply set.eq_of_subset_of_subset,
+  { change σ (0 : A) with (resolvent 𝕜 (0 : A))ᶜ,
+    rw set.compl_subset_comm,
+    intros k hk,
+    rw set.mem_compl_singleton_iff at hk,
+    have unit_k_one : is_unit (units.mk0 k hk • (1 : A)),
+      from is_unit.smul (units.mk0 k hk) is_unit_one,
+    rw mem_resolvent_iff,
+    simpa [algebra.algebra_map_eq_smul_one], },
+  { rw [set.singleton_subset_iff, mem_iff],
+    simp [algebra.algebra_map_eq_smul_one], },
```

Comment:
```suggestion
  refine set.eq_of_subset_of_subset _ (by simp [algebra.algebra_map_eq_smul_one, mem_iff]),
  rw [spectrum, set.compl_subset_comm],
  intros k hk,
  rw set.mem_compl_singleton_iff at hk,
  have : is_unit (units.mk0 k hk • (1 : A)) := is_unit.smul (units.mk0 k hk) is_unit_one,
  simpa [mem_resolvent_iff, algebra.algebra_map_eq_smul_one]
```

🏌️, but more importantly I think it's advocated to avoid `change` when possible, as it can be a bit brittle. also, I think the more common spelling of that first lemma is `set.subset.antisymm`, but it's fine either way

Reply:
good to know about `change`